src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/UserManagementServicePermissionTests.cs

```diff
@@ -0,0 +1,161 @@
+                        {
+                            Email = userEmail,
+                            Accepted = false,
+                            Role = new IdentityRole(),
+                            RoleId = Guid.Empty.ToString(),
+                            Created = DateTime.UtcNow,
+                            CreatedBy = "Test creator"
+                        };
+
+                        var userAndRolesContextId = Guid.NewGuid().ToString();
+                        await using (var userAndRolesDbContext =
+                            DbUtils.InMemoryUserAndRolesDbContext(userAndRolesContextId))
+                        {
+                            await userAndRolesDbContext.AddAsync(userInvite);
+                            await userAndRolesDbContext.SaveChangesAsync();
+                        }
+
+                        await using (var userAndRolesDbContext =
+                            DbUtils.InMemoryUserAndRolesDbContext(userAndRolesContextId))
+                        {
+                            var userManagementService = BuildUserManagementService(userAndRolesDbContext, userService.Object);
+                            return await userManagementService.CancelInviteAsync(userEmail);
+                        }
+                    });
+        }
+
+        private static Mock<IConfiguration> ConfigurationMock()
+        {
+            var templateSection = new Mock<IConfigurationSection>();
+            var adminUriSection = new Mock<IConfigurationSection>();
+            var configuration = new Mock<IConfiguration>();
+
+            templateSection.Setup(m => m.Value)
+                .Returns("the-template-id");
+
+            adminUriSection.Setup(m => m.Value)
+                .Returns("admin-uri");
+
+            configuration
+                .Setup(m => m.GetSection("NotifyInviteTemplateId"))
+                .Returns(templateSection.Object);
+
+            configuration
+                .Setup(m => m.GetSection("AdminUri"))
+                .Returns(adminUriSection.Object);
+
+            return configuration;
+        }
+
+        private static UserManagementService BuildUserManagementService(
+            UsersAndRolesDbContext usersAndRolesDbContext = null,
+            IUserService userService = null,
+            ContentDbContext contentDbContext = null,
+            IEmailService emailService = null,
+            IConfiguration configuration = null,
+            UserManager<ApplicationUser> userManager = null,
+            IPersistenceHelper<ContentDbContext> persistenceHelper = null)
+        {
+            return new UserManagementService(
+                usersAndRolesDbContext ?? new Mock<UsersAndRolesDbContext>().Object,
+                userService ?? MockUtils.AlwaysTrueUserService().Object,
+                contentDbContext ?? new Mock<ContentDbContext>().Object,
+                emailService ?? new Mock<IEmailService>().Object,
+                configuration,
```

Comment:
Any reason why not to provide non-null defaults for these final 3 arguments, as in the 4 services above?

Reply:
Talked about it on Slack - added default arguments for most and use ConfigurationMock() as the default for configuration.